massa-network-exports/src/network_controller.rs

```diff
@@ -213,18 +213,20 @@ impl NetworkCommandSender {
     /// # Returns
     /// Can return a `[NetworkError::ChannelError]` that must be managed by the direct caller of the
     /// function.
-    pub async fn send_ask_for_operations(
+    pub fn send_ask_for_operations(
         &self,
         to_node: NodeId,
         wishlist: OperationIds,
     ) -> Result<(), NetworkError> {
-        self.0
-            .send(NetworkCommand::AskForOperations { to_node, wishlist })
-            .await
-            .map_err(|_| {
-                NetworkError::ChannelError("could not send AskForOperations command".into())
-            })?;
-        Ok(())
+        match self.0.try_reserve() {
+            Ok(permit) => {
+                permit.send(NetworkCommand::AskForOperations { to_node, wishlist });
+                Ok(())
+            }
+            Err(_) => Err(NetworkError::ChannelError(
+                "Failed to acquire permit to send AskForOperations command".into(),
+            )),
+        }
```

Comment:
I didn't follow well, what is the enhancement in comparison to the previous timeout?

Reply:
The idea is that whenever the channel is full, we stop trying to send anything further, so that instead protocol can handle messages from network(since the channel is supposedly full because network is trying to send a message to protocol).